src/FpdiProtection.php

```diff
@@ -193,26 +193,45 @@ class FpdiProtection extends \setasign\Fpdi\Fpdi
      */
     protected $fileIdentifier;
 
+    /**
+     * @var bool
+     */
+    protected $useArcfourFallback;
+
     /**
      * FpdiProtection constructor.
      *
      * @param string $orientation
      * @param string $unit
      * @param string $size
+     * @param bool $useArcfourFallback
      */
-    public function __construct($orientation = 'P', $unit = 'mm', $size = 'A4')
+    public function __construct($orientation = 'P', $unit = 'mm', $size = 'A4', $useArcfourFallback = false)
     {
         parent::__construct($orientation, $unit, $size);
 
         $randomBytes = function_exists('random_bytes') ? \random_bytes(32) : \mt_rand();
         $this->fileIdentifier = md5(__FILE__ . PHP_SAPI . PHP_VERSION . $randomBytes, true);
+        $this->useArcfourFallback = $useArcfourFallback;
+
+        if($useArcfourFallback) return;
 
-        if (!function_exists('openssl_encrypt') || !in_array('rc4-40', openssl_get_cipher_methods(), true)) {
+        if(OPENSSL_VERSION_NUMBER >= 0x30000000 && (PHP_VERSION_ID < 80200 || PHP_VERSION_ID >= 70100)) {
```

Comment:
Isn't the "OpenSSL 3" simply incompatible to PHP versions below 8.1? Why a do you check for 8.2 and 7.1? See https://www.php.net/manual/en/openssl.requirements.php ?

Please also add a space: `if (OPENSSL_...`.